### PR TITLE
Mxgda 3733: Minimal gridscan plan

### PR DESCRIPTION
**Pull request by DominicOram**
Fixes https://jira.diamond.ac.uk/browse/MXGDA-3733 and https://jira.diamond.ac.uk/browse/MXGDA-3765.

Note, is branched from https://github.com/DiamondLightSource/python-artemis/pull/9 so merge that first.

To test:
* Cherrypick the code in https://github.com/DiamondLightSource/python-artemis/tree/test_minimal_scan (this is required due to https://github.com/DiamondLightSource/python-artemis/issues/13 and https://github.com/DiamondLightSource/python-artemis/issues/11)
* Run this code i.e. `pipenv run python src/artemis/fast_grid_scan_plan.py`
* Confirm that the coe runs without any errors and gives a progress bar of what it's doing
* Confirm logs are produced in `/tmp/bluesky.log` and `/tmp/ophyd.log`

### CODE REVIEW

**Comment by DominicOram** [comment]
Once this code is merged delete https://github.com/DiamondLightSource/python-artemis/tree/test_minimal_scan

File: src/artemis/fast_grid_scan_plan.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    x_start=0.0,
+    y1_start=0.0,
+    z1_start=0.0,
+)
+DETECTOR_PARAMS = DetectorParams(
+    current_energy=100,
+    exposure_time=0.1,
+    acquisition_id="test",
+    directory="/tmp",
+    prefix="file.h5",
```
Comment (bentom08) [inline src/artemis/fast_grid_scan_plan.py:59]: If, as I assume, these are intended as example parameters the extension should be removed, as it's incorrect and is added on by odin anyway

**Review by bentom08 — CHANGES_REQUESTED** [review]
Looks good, just a minor, slightly picky comment on the parameters

File: src/artemis/fast_grid_scan_plan.py (hunk 1 above)
Comment (DominicOram) [reply src/artemis/fast_grid_scan_plan.py]: Agreed, fixed

**Pull request merged**